### PR TITLE
metrics: add statement pessimistic retry count

### PR DESCRIPTION
**Pull request by coocood**
### What problem does this PR solve? 
add statement pessimistic retry count

### What is changed and how it works?
add statement pessimistic retry count

### Check List 

Tests 

 - No code

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by coocood** [comment]
@jackysp @cfzjywxk PTAL

File: executor/adapter.go [hunk 1]
```diff
@@ -286,6 +286,9 @@ func (a *ExecStmt) Exec(ctx context.Context) (_ sqlexec.RecordSet, err error) {
 	defer func() {
 		r := recover()
 		if r == nil {
+			if a.retryCount > 0 {
+				metrics.StatementPessimisticRetryCount.Observe(float64(a.retryCount))
```
Comment (cfzjywxk) [inline executor/adapter.go]: Do we need to group by `err` results nil or not?

**Comment by coocood** [comment]
@cfzjywxk PTAL

**Comment by cfzjywxk** [comment]
LGTM

**Pull request merged**